src/window.cc

```diff
@@ -165,23 +166,94 @@ void Window::create_menu() {
+    auto time_now=std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+      boost::filesystem::path path = Dialog::new_folder();
+      if(boost::filesystem::exists(path)) {
+        if(boost::filesystem::last_write_time(path)>=time_now) {
+          if(Singleton::directories()->current_path!="")
+            Singleton::directories()->update();
+          Singleton::terminal()->print("New folder "+path.string()+" created.\n");
+        }
+        else
+          Singleton::terminal()->print("Error: "+path.string()+" already exists.\n");
+      } else {
+        Singleton::terminal()->print("Cancel \n");
+      }
+      Singleton::directories()->select(path);
   });
   menu.action_group->add(Gtk::Action::create("FileNewProject", "New Project"));
   menu.action_group->add(Gtk::Action::create("FileNewProjectCpp", "C++"), [this]() {
-    new_cpp_project_dialog();
+      boost::filesystem::path project_path = Dialog::new_folder();
+      auto project_name=project_path.filename().string();
+      for(size_t c=0;c<project_name.size();c++) {
+        if(project_name[c]==' ')
+          project_name[c]='_';
+      }
+      auto cmakelists_path=project_path;
+      cmakelists_path+="/CMakeLists.txt";
+      auto cpp_main_path=project_path;
+      cpp_main_path+="/main.cpp";
+      if(boost::filesystem::exists(cmakelists_path)) {
+        Singleton::terminal()->print("Error: "+cmakelists_path.string()+" already exists.\n");
+        return;
+      }
+      if(boost::filesystem::exists(cpp_main_path)) {
+        Singleton::terminal()->print("Error: "+cpp_main_path.string()+" already exists.\n");
+        return;
+      }
+      std::string cmakelists="cmake_minimum_required(VERSION 2.8)\n\nproject("+project_name+")\n\nset(CMAKE_CXX_FLAGS \"${CMAKE_CXX_FLAGS} -std=c++1y -Wall\")\n\nadd_executable("+project_name+" main.cpp)\n";
+      std::string cpp_main="#include <iostream>\n\nusing namespace std;\n\nint main() {\n  cout << \"Hello World!\" << endl;\n\n  return 0;\n}\n";
+      if(juci::filesystem::write(cmakelists_path, cmakelists) && juci::filesystem::write(cpp_main_path, cpp_main)) {
+        Singleton::directories()->open(project_path);
+        notebook.open(cpp_main_path);
+        Singleton::terminal()->print("C++ project "+project_name+" created.\n");
+      }
+      else
+        Singleton::terminal()->print("Error: Could not create project "+project_path.string()+"\n");
   });
   menu.action_group->add(Gtk::Action::create("FileOpenFile", "Open File"), Gtk::AccelKey(menu.key_map["open_file"]), [this]() {
-    open_file_dialog();
+      notebook.open(Dialog::select_file());
   });
   menu.action_group->add(Gtk::Action::create("FileOpenFolder", "Open Folder"), Gtk::AccelKey(menu.key_map["open_folder"]), [this]() {
-    open_folder_dialog();
+    auto path = Dialog::select_folder();
+    if (boost::filesystem::exists(path))
+      Singleton::directories()->open(path);
+    else
+      Singleton::terminal()->print("Cancel \n");
```

Comment:
I will expand the functionality of the dlls to discover cancel, X, esc etc. This is basically an quickfix.